bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/resource/GraphqlResourceProvider.java

```diff
@@ -53,6 +53,11 @@
     public Resource getResource(ResolveContext<T> ctx, String path, ResourceContext resourceContext, Resource parent) {
         LOGGER.debug("getResource called for " + path);
 
+        // needed for testing, harmless during runtime
+        if (path.endsWith("/.")) {
```

Comment:
Is this really needed? Looks odd to me to check for a path *ending* with `/.` (while your PR fixes property names *starting* with `./`.

Reply:
I agree. Mixing production and test code is a bad practice and should be avoided.